app/javascript/controllers/participants_controller.js

```diff
@@ -0,0 +1,53 @@
+import { Controller } from '@hotwired/stimulus'
+
+export default class extends Controller {
+  static targets = ['input', 'invalidFeedbackContainer', 'invalidFeedbackTemplate']
+
+  static values = { url: String }
+
+  static outlets = ['nested-form']
+
+  async lookup (event) {
+    event.preventDefault()
+
+    if (this.inputTarget.value === '') return
+
+    this.invalidFeedbackContainerTarget.innerHTML = ''
+    const ids = this.inputTarget.value.split(/[,;]? |\n/)
+    const errorIds = await Promise.all(ids.map(id => this.lookupId(id)))
+    this.inputTarget.value = errorIds.filter(id => id).join('\n')
+  }
+
+  async lookupId (id) {
+    const sunetid = id.replaceAll('@stanford.edu', '')
+    return fetch(`${this.urlValue}?id=${sunetid}`)
+      .then(response => {
+        if (response.status === 404) return null
+        if (!response.ok) throw new Error(response.status)
+        return response.json()
+      })
+      .then(data => {
+        if (!data) {
+          this.addError(id)
+          return id
+        }
+        this.addParticipant(data)
+        return null
+      })
+      .catch(error => console.dir(error))
```

Comment:
Is this worth honeybadgering or not so much?

Reply:
We haven't HBed from JS before when accessing our own APIs. I'd say "no" for now, but "definitely" if we encounter any problems.